x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/ApiKey.java

```diff
@@ -153,6 +160,14 @@ public ApiKey(StreamInput in) throws IOException {
             this.name = in.readString();
         }
         this.id = in.readString();
+        if (in.getTransportVersion().onOrAfter(TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCR)) {
```

Comment:
We may need a new TransportVersion for this given the recent changes to versioning. I am reaching out to the experts for confirmation.

Reply:
A new TransportVersion is added https://github.com/elastic/elasticsearch/pull/96014/commits/f3c58f7ffff972da2c0deabd088008ce1464e4b6